src/libraries/System.Private.CoreLib/src/System/Globalization/CalendarData.Icu.cs

```diff
@@ -434,7 +434,12 @@ internal static unsafe bool EnumCalendarInfo(string localeName, CalendarId calen
 #pragma warning disable CS8500 // takes address of managed type
         private static unsafe bool EnumCalendarInfo(string localeName, CalendarId calendarId, CalendarDataType dataType, IcuEnumCalendarsData* callbackContext)
         {
+#if TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS
+            callbackContext->Results.AddRange(GetCalendarInfoNative(localeName, calendarId, dataType).Split("||"));
+            return callbackContext->Results.Count > 0;
```

Comment:
It might be more consistent to do it in `internal static unsafe bool EnumCalendarInfo(string localeName, CalendarId calendarId, CalendarDataType dataType, out string[]? calendarData)`. Correct me if I understand it wrong:
- We have `calendarData` that an array there, checking its length is more straightforward.
- We can avoid `AddRange`, we can allocate the array with the suitable size at once.
- It is fine to have the line `callbackContext->Results.AddRange(GetCalendarInfoNative(localeName, calendarId, dataType).Split("||"));` but it is a bit more readable if we split it into 2 commands with a middle-named value:
```
string[] calendarInfo = GetCalendarInfoNative(localeName, calendarId, dataType).Split("||");
callbackContext->Results.AddRange(calendarInfo);
```

Reply:
> It might be more consistent to do it in internal static unsafe bool EnumCalendarInfo(string localeName, CalendarId calendarId, CalendarDataType dataType, out string[]? calendarData)

I agree. I've chosen this way because I didn't want to modify the `EnumMonthNames` and `EnumDatePatterns` functions. If you look at `EnumMonthNames` and `EnumDatePatterns, they call directly `EnumCalendarInfo(string localeName, CalendarId calendarId, CalendarDataType dataType, IcuEnumCalendarsData* callbackContext)` instead of going thru `EnumCalendarInfo(string localeName, CalendarId calendarId, CalendarDataType dataType, out string[]? calendarData)`.

While refactoring the `EnumMonthNames` should be straightforward, the `EnumDatePatterns` sets `callbackContext.DisallowDuplicates = true` on callback.

Let me know what you think!

Reply:
I see, in this situation you are right, let's leave it as it is.